### ISSUE TITLE
Pingbacks slipping through

### ISSUE THREAD

**Issue by kochs-online** [dartiss/no-self-ping#7]
I run a self-hosted multi-site installation of WordPress and have activated this plugin network-wide for some years. Recently I noticed in one of the blogs that there are pingbacks to the same blog. While some of the links _did_ ping, others _didn't_ though – even within the same blog post.
As I'm not the author of this blog I can only speculate about what happened, so this is more of a question: What happens, if I'm linking to https resources while the home URL uses http? What about relative links?

**Comment by dartiss** [comment]
Hi.

Relative links won't work because it performs a match on the URL, as defined in the settings. However, the recent versions added a text box that allows you to add other URLs as well, so if the SSL status is causing problems, you can add the alternative URL into that.

**Comment by kochs-online** [comment]
Yes, I figured that out, but from a user's perspective I wouldn't expect the scheme to be of relevance.

**Comment by kochs-online** [comment]
I think I'd leave out the scheme completely. That way even schemeless URIs get treated correctly.
First I'd merge `$home` and `$url_array` (to simplify the code) and then strip off the scheme with this regex:
`^[[:alpha:]][[:alnum:]\.\-\+]*:`

**Comment by dartiss** [comment]
I understand. I avoided this before as switching out the scheme does leave the possibility of a similar named site being caught. It's highly unlikely though.

Yeah, leave this with me and I'll implement something on the next iteration. Thanks for suggestions and clarifications.

**Comment by kochs-online** [comment]
Great! If you need assistance, I'd be happy to help.

I agree, increasing the complexity also increases the risk of false results. Never used them in PHP, but maybe unit tests are the way to go here?

Testing the regex above I used this set of URIs:
```
http://www.example.com
http://www.example.com/
http://www.example.com/index.html
https://www.example.com/index.html
a.1-some+strange+scheme://www.example.com/index.html
http://www.example.com/example.com/index.html
//www.example.com/index.html
/absolute/index.html
relative/index.html
http://example.com/index.html
http://www2.example.com/index.html
http://www2.example.com/index.html?url=http://www.example.com/
```

**Comment by dartiss** [comment]
In #13 I'm auto generating the site list from now, curating sub-sites URL in MU set-ups. As part of this, I'm stripping the scheme and then adding 2 records for each site, one SSL and one not. This should pick up the instances that you raised.
